hikari/internal/routes.py

```diff
@@ -214,10 +214,15 @@ def _(self, _: attrs.Attribute[typing.AbstractSet[str]], values: typing.Abstract
             msg = f"{self.path_template} must have at least one valid format set"
             raise ValueError(msg)
 
-    is_sizable: bool = attrs.field(default=True, kw_only=True, repr=False, hash=False, eq=False)
-    """Whether a `size` param can be specified."""
-
-    def compile(self, base_url: str, *, file_format: str, size: int | None = None, **kwargs: object) -> str:
+    def compile(
+        self,
+        base_url: str,
+        *,
+        file_format: str,
+        size: int | None = None,
+        settings: dict[str, str | bool | None] | None = None,
```

Comment:
```suggestion
        size: int = None,
        lossless: undefined.UndefinedOr[bool] = undefined.UNDEFINED,
```

+ all the other ones